app/models.py

```diff
@@ -2551,3 +2551,90 @@ def serialize(self):
+        min(created_at) as earliest_timestamp,
+        max(created_at) as latest_timestamp,
+        (case when downloaded_at is null then false else true end) as has_been_downloaded
+        count(*)
+    FROM unsubscribe_requests
+    group by service_id, unsubscribe_request_report_id
+    """
+
+    # Tell alembic not to create this as a table. We have a migration where we manually set this up as a view.
+    # This is custom logic we apply - not built-in logic. See `migrations/env.py`
+    __table_args__ = {"info": {"managed_by_alembic": False}}
+
+    id = db.Column(UUID(as_uuid=True), primary_key=True)
+    service_id = db.Column(UUID(as_uuid=True), db.ForeignKey("services.id"), nullable=False)
+    service = db.relationship(Service, backref=db.backref("unsubscribe_requests"))
+
+    earliest_timestamp = db.Column(db.DateTime, nullable=False, default=datetime.datetime.utcnow)
+    latest_timestamp = db.Column(db.DateTime, nullable=False, default=datetime.datetime.utcnow)
+    has_been_downloaded = db.Column(db.Boolean, nullable=False)
+    count = db.Column(db.Integer, nullable=False)
+
+
+class UnsubscribeRequest(db.Model):
+    __tablename__ = "unsubscribe_request"
+
+    id = db.Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
+
+    # WIP:
+    # Ignoring a strict foreign key relationship here for now. Notifications are archived to the NotificationHistory
+    # table by a nightly job and I haven't investigated whether that might break a strict FK yet or if it would
+    # work smoothly. We can still have a relationship using an explicit join condition.
+    notification_id = db.Column(UUID(as_uuid=True), index=True, nullable=False)
+    notification = db.relationship(
+        "NotificationAllTimeView",
+        primaryjoin="NotificationLetterDespatch.notification_id == foreign(NotificationAllTimeView.id)",
+        uselist=False,
+    )
+
+    # this is denormalised but might still be useful to have as a separate column?
+    service_id = db.Column(UUID(as_uuid=True), db.ForeignKey("services.id"), nullable=False)
+    service = db.relationship(Service, backref=db.backref("unsubscribe_requests"))
+
+    # is it useful to
+    template_id = db.Column(UUID(as_uuid=True), db.ForeignKey("templates.id"), nullable=False)
+    template_version = db.Column(db.Integer)
+    template = db.relationship(Template, backref=db.backref("unsubscribe_requests"))
+    template_history = db.relationship(TemplateHistory, backref=db.backref("unsubscribe_requests"))
+
+    # nullable as at some point we may want to delete these for data retention purposes
+    email_address = db.Column(db.Text, nullable=True)
+
+    created_at = db.Column(db.DateTime, nullable=False, default=datetime.datetime.utcnow)
+
+    # is this useful or is a boolean enough? if we create a table for a report, we could remove from this table entirely
+    downloaded_at = db.Column(db.DateTime, nullable=False, default=datetime.datetime.utcnow)
```

Comment:
I think `downloaded_at` should be removed from this table entirely as we already have `has_been_downloaded` in the unsubscribe_request_report table and we know the unsubscribe request will be downloaded as part of the report.

Reply:
i've renamed `downloaded_at` to `processed_at` - i'd slightly misinterpreted the slides and had got downloading vs processing mixed up in my head. i dont think its useful to record when something is downloaded, only when it's processed.

Reply:
so my approach involved using a view for `unsubscribe_request_report` rather than a table.

So the report doesn't actually exist, and there's no value there to be set - it's just derived by looking at this value

Reply:
resolving as no longer a view. also changed downloaded_at, thanks for the suggestion!